### PR TITLE
Correlation Dashboard

### PR DESCRIPTION
**Pull request by colin99d**
A new dashboard that allows people to see the correlation between different stock's various features for different periods of time. Also adds a function that makes adding future dashboards simpler.

### CODE REVIEW

File: gamestonk_terminal/jupyter/dashboards/correlation.ipynb [hunk 1]
```diff
@@ -0,0 +1,199 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 3,
```
Comment (piiq) [inline gamestonk_terminal/jupyter/dashboards/correlation.ipynb:5]: shall we clear the notebook outputs and metadata by default? i can try to add it to the pre-commit hook

File: gamestonk_terminal/jupyter/dashboards/dashboards_controller.py [hunk 2]
```diff
@@ -37,49 +35,60 @@ def print_help(self):
...
+        help="Shows dashboard in jupyter-lab.",
+    )
 
+    ns_parser = parse_known_args_and_warn(
+        parser, other_args, EXPORT_ONLY_RAW_DATA_ALLOWED
+    )
+
+    if ns_parser:
+        cmd = "jupyter-lab" if ns_parser.jupyter else "voila"
+        file = f"gamestonk_terminal/jupyter/dashboards/{filename}.ipynb"
```
Comment (piiq) [inline gamestonk_terminal/jupyter/dashboards/dashboards_controller.py:75]: This path will probably not work on Windows.
You can use `os.path` or `pathlib` to construct the paths in an OS-agnostic way (search for `os.path.join` in the repo for examples)

**Review by piiq — CHANGES_REQUESTED** [review]
🔥 Dashboards are cool. 
I've left tiny comments in the code

File: gamestonk_terminal/jupyter/dashboards/correlation.ipynb (hunk 1 above)
Comment (colin99d) [reply gamestonk_terminal/jupyter/dashboards/correlation.ipynb:5]: That would be very helpful. I already forgot that little line of code you sent me, but having it automatically run would be tremendous.

Comment (colin99d) [reply gamestonk_terminal/jupyter/dashboards/correlation.ipynb]: @piiq I added a pre-commit hook to remove kernel metadata.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -30,6 +30,14 @@ repos:
       - id: mypy
         args: ['--ignore-missing-imports', '--exclude=/setup\.py$']
         additional_dependencies: [types-all]
+
+  - repo: https://github.com/roy-ht/pre-commit-jupyter
+    rev: v1.2.1
+    hooks:
+      - id: jupyter-notebook-cleanup
+        args:
+          # - --remove-kernel-metadata
```
Comment (piiq) [inline .pre-commit-config.yaml:39]: can you try to uncomment this line also?

Comment (colin99d) [reply .pre-commit-config.yaml]: Done!

**Review by piiq — APPROVED** [review]
It works in voila
image


and in jupyter lab
image

**Pull request merged**